datafusion/sqllogictest/test_files/cte.slt

```diff
@@ -122,6 +110,22 @@ physical_plan
 08)----------RepartitionExec: partitioning=RoundRobinBatch(4), input_partitions=1
 09)------------WorkTableExec: name=nodes
 
+# trivial deduplicating recursive CTE works
+query I
+WITH RECURSIVE nodes AS (
+    SELECT 1 as id
+    UNION ALL
```

Comment:
I think this was meant to be `UNION`.

And the original test wasn't generating duplicate results with `UNION ALL`.
Maybe we should use a different query, I could think of some thing like this, 
but I believe there should be a better one.

```sql
WITH RECURSIVE nodes AS (
    SELECT id from (VALUES (1), (2)) nodes(id)
    UNION
    SELECT id + 1 as id
    FROM nodes
    WHERE id < 4
)
SELECT * FROM nodes;
```